packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -89,10 +92,40 @@ export async function streamExtractExternal(
         clearTimeout(timer)
         reject(error)
       })
-      .pipe(unzip.Extract({path: directory}))
-      .on('close', () => {
+      .pipe(unzip.Parse())
+      .on('entry', (entry: unzip.Entry) => {
+        const entryPath = path
+          .normalize(entry.path)
+          .replace(/^(\.\.(\/|\\|$))+/, '')
```

Comment:
What if entryPath is empty after replace is called?

Reply:
For example, the regexp pattern appears to be designed to handle cases like:

- `..`
- `../`
- `..\`

which will all result in an empty string after the replace call

Reply:
`path.join` ignores zero-length paths: https://nodejs.org/api/path.html#pathjoinpaths